shared/contracts/BaseTemplate.sol

```diff
@@ -0,0 +1,191 @@
+        acl.createPermission(grantee, finance, finance.EXECUTE_PAYMENTS_ROLE(), manager);
+        acl.createPermission(grantee, finance, finance.MANAGE_PAYMENTS_ROLE(), manager);
+    }
+
+    /* TOKEN MANAGER */
+
+    function _installTokenManagerApp(Kernel dao, MiniMeToken token, bool transferable, uint256 maxAccountTokens) internal returns (TokenManager) {
+        TokenManager tokenManager = TokenManager(_installNonDefaultApp(dao, TOKEN_MANAGER_APP_ID));
+        token.changeController(tokenManager);
+        tokenManager.initialize(token, transferable, maxAccountTokens);
+        return tokenManager;
+    }
+
+    function _createTokenManagerPermissions(ACL acl, TokenManager tokenManager, address grantee, address manager) internal {
+        acl.createPermission(grantee, tokenManager, tokenManager.MINT_ROLE(), manager);
+        acl.createPermission(grantee, tokenManager, tokenManager.ASSIGN_ROLE(), manager);
+        acl.createPermission(grantee, tokenManager, tokenManager.REVOKE_VESTINGS_ROLE(), manager);
+    }
+
+    /* VAULT */
+
+    function _installVaultApp(Kernel dao) internal returns (Vault) {
+        Vault vault = Vault(_installDefaultApp(dao, VAULT_APP_ID));
+        vault.initialize();
+        return vault;
+    }
+
+    function _createVaultPermissions(ACL acl, Vault vault, address grantee, address manager) internal {
+        acl.createPermission(grantee, vault, vault.TRANSFER_ROLE(), manager);
+    }
+
+    /* VOTING */
+
+    function _installVotingApp(Kernel dao, MiniMeToken token, uint64 support, uint64 acceptance, uint64 duration) internal returns (Voting) {
+        Voting voting = Voting(_installNonDefaultApp(dao, VOTING_APP_ID));
+        voting.initialize(token, support, acceptance, duration);
+        return voting;
+    }
+
+    function _createVotingPermissions(ACL acl, Voting voting, address grantee, address manager) internal {
+        acl.createPermission(grantee, voting, voting.MODIFY_QUORUM_ROLE(), manager);
+        acl.createPermission(grantee, voting, voting.MODIFY_SUPPORT_ROLE(), manager);
+    }
+
+    /* EVM SCRIPTS */
+
+    function _createEvmScriptsRegistryPermissions(ACL acl, address grantee, address manager) internal {
+        EVMScriptRegistry registry = EVMScriptRegistry(acl.getEVMScriptRegistry());
+        acl.createPermission(grantee, registry, registry.REGISTRY_MANAGER_ROLE(), manager);
+        acl.createPermission(grantee, registry, registry.REGISTRY_ADD_EXECUTOR_ROLE(), manager);
+    }
+
+    /* APPS */
+
+    function _installNonDefaultApp(Kernel dao, bytes32 appId) internal returns (address) {
+        return _installApp(dao, appId, new bytes(0), false);
+    }
+
+    function _installDefaultApp(Kernel dao, bytes32 appId) internal returns (address) {
+        return _installApp(dao, appId, new bytes(0), true);
+    }
+
+    function _installApp(Kernel dao, bytes32 appId, bytes data, bool setDefault) internal returns (address) {
```

Comment:
Perhaps we can rename this `_installAppFromAPM()`.

Also, what do you think about directly using this in most places, at least in the base template (since we don't need great UX ourselves)? We'd optimize the base contract's size, and could save a bit of gas by directly doing the `initialize()` via the `data` too.